libcxx/include/string

```diff
@@ -1874,8 +1874,6 @@ private:
   operator==(const basic_string<char, char_traits<char>, _Alloc>& __lhs,
```

Comment:
Not attached: Let's not mark this change as a NFC, since it's definitely not doing "nothing".